rfcs/text/0000-alternative-planning.md

```diff
@@ -0,0 +1,315 @@
+in the the DCP. It then proposes alternative approaches to the problems that this RFC is looking to tackle.
+
+Please note: the list of reviewers chosen is somewhat arbitrary based on my memory of who concerns themselves with this
+stuff. And Github limits the number to 15. Apologies if I left you out - guaranteed I'd like you to read this even/especially 
+if you strongly disagree.
+
+## Author(s)
+
+[Justin Clark-Casey](mailto:justincc@ebi.ac.uk)
+
+## Shepherd
+***Leave this blank.*** This role is assigned by DCP PM to guide the **Author(s)** through the RFC process.*
+
+*Recommended format for Shepherds:*
+
+ `[Name](mailto:username@example.com)`
+
+## Motivation
+
+Roadmaps and planning are the crankshaft of the DCP. If we don't create a crankshaft suitable for the engine of development 
+then the DCP will run inefficiently or not at all. I'm concerned that the proposed process for creating these roadmaps and plans, as 
+described in the Roadmaps and Planning RFC and as an evolution of the current process, will not fit DCP organizational and development realities.
+
+This RFC tries to concisely describe those problems and propose alternative solutions. It comes as a result of being
+asked to write these alternatives up. Due to time constraints it's 
+nowhere near as comprehensive, well-written or referenced as I would like. Nonetheless, I hope it can help spur 
+constructive discussion.
+
+This RFC is the author's own point of view as a technical lead and someone with long experience in highly distributed 
+open-source projects that require co-operation between peer organizations and individuals.
+
+## Detailed Design
+
+From my understanding, the roadmaps and planning RFC describes a 70 day process where scientists, project leads, product owners, user experience
+experts and technical leads create a roadmap consisting of themes, each theme having one of more objectives. These 
+objectives are further described by user stories. Each objective is represented in the DCP Zenhub board as an epic with
+user stories represented by an item attached to the epic. Further items can be created as spikes to spend time clarifying
+the user stories and/or performing design work prior to implementation. Each item is assigned a milestone where there are
+3 milestones of equal length in a quarter.
+
+In my experience of the existing planning process almost all spikes are cross-component issues. DCP culture pretty much
+ mandates that these require an RFC. An RFC
+takes a minimum of 3 weeks from inception to approval, though realistically RFCs of any complexity require considerably
+more time to understand the problem and align with other planned and existing technical work.
+
+From my experience so far, if an RFC is required this means that it will take a minimum of 2 milestones (8 weeks)
+before it can be considered complete. A T-shirt estimate is only possible once the RFC is approved and there is a
```

Comment:
There is nothing preventing teams from writing smaller RFC(s) and iterating towards an objective similar to AUDR1 and AUDR2. As needed, smaller scope should be negotiated to ensure that we're not spending 8 weeks on RFCs or slipping a Spike because we need one more RFC than originally planned.

Reply:
The point is that writing RFCs is a big overhead, there is little incentive to do smaller ones. And once you include negotiation that's another cross-DCP round of discussion which makes the process take even longer.

Reply:
it would be worthwhile looking into why RFCs are such a big overhead. Is it the process or the nature of the knowledge we are trying to communicate?

Personally, I find the lack of overall context makes the process more difficult.  One either has to write up a lot of context or reviews have to have that context.  If RFCs were pull requests against as system architecture or component API documentation, it would be much lighter weight.

Architectural Decisions Records (ADR) might be a lighter-weight approach.  I don't have experience with them, so I can't say much beyond what I read on the internet.

https://niels.nu/blog/2018/lightweight-architecture-decision-records.html